src/main/java/ch/njol/skript/sections/EffSecShoot.java

```diff
@@ -0,0 +1,214 @@
+			"(make|let) %livingentities/locations% shoot %entitydatas% [(at|with) (speed|velocity) %-number%] [%-direction%]");
+		EventValues.registerEventValue(ShootEvent.class, Entity.class, new Getter<Entity, ShootEvent>() {
+			@Override
+			public Entity get(ShootEvent shootEvent) {
+				return shootEvent.getEntity();
+			}
+		}, 0);
+	}
+
+	private final static Double DEFAULT_SPEED = 5.;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<EntityData<?>> types;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<?> shooters;
+	@Nullable
+	private Expression<Number> velocity;
+	@Nullable
+	private Expression<Direction> direction;
+
+	@Nullable
+	public static Entity lastSpawned = null;
+
+	@Nullable
+	private Trigger trigger;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs,
+						int matchedPattern,
+						Kleenean isDelayed,
+						SkriptParser.ParseResult parseResult,
+						@Nullable SectionNode sectionNode,
+						@Nullable List<TriggerItem> triggerItems) {
+		types = (Expression<EntityData<?>>) exprs[matchedPattern];
+		shooters = exprs[1 - matchedPattern];
+		velocity = (Expression<Number>) exprs[2];
+		direction = (Expression<Direction>) exprs[3];
+
+		if (sectionNode != null) {
+			trigger = loadCode(sectionNode, "shoot", ShootEvent.class);
+		}
+
+		return true;
+	}
+
+	@Override
+	@Nullable
+	@SuppressWarnings({"unchecked", "rawtypes"})
+	protected TriggerItem walk(Event e) {
+		lastSpawned = null;
+
+		Object localVars = Variables.copyLocalVariables(e);
+
+		Consumer<? extends Entity> consumer;
+		if (trigger != null) {
+			consumer = o -> {
+				lastSpawned = o;
+				ShootEvent shootEvent = new ShootEvent(o);
+				// Copy the local variables from the calling code to this section
+				Variables.setLocalVariables(shootEvent, localVars);
+				trigger.execute(shootEvent);
+			};
+		} else {
+			consumer = null;
+		}
+
+		Number v = velocity != null ? velocity.getSingle(e) : DEFAULT_SPEED;
+		if (v != null){
+			Direction dir = direction != null ? direction.getSingle(e) : Direction.IDENTITY;
+			if (dir != null){
+				for (Object shooter : shooters.getArray(e)) {
+					for (EntityData<?> d : types.getArray(e)) {
+						if (shooter instanceof LivingEntity) {
+							Vector vel = dir.getDirection(((LivingEntity) shooter).getLocation()).multiply(v.doubleValue());
+							Class<? extends Entity> type = d.getType();
+							if (Fireball.class.isAssignableFrom(type)) {// fireballs explode in the shooter's face by default
+								Fireball projectile = (Fireball) d.spawn(((LivingEntity) shooter).getEyeLocation().add(vel.clone().normalize().multiply(0.5)), (Consumer) consumer);
+								projectile.setShooter((ProjectileSource) shooter);
+								projectile.setVelocity(vel);
+								lastSpawned = projectile;
+							} else if (Projectile.class.isAssignableFrom(type)) {
+								@SuppressWarnings("unchecked")
```

Comment:
Shouldn't be needed since you suppress this for the whole method

We also don't use suppressions for individual lines